plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/Maven2MavenPathParser.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal.maven2;
+
+import org.sonatype.nexus.proxy.maven.gav.Gav;
+import org.sonatype.nexus.proxy.maven.gav.M2GavCalculator;
+import org.sonatype.nexus.repository.maven.internal.MavenPath;
+import org.sonatype.nexus.repository.maven.internal.MavenPath.Coordinates;
+import org.sonatype.nexus.repository.maven.internal.MavenPath.SignatureType;
+import org.sonatype.nexus.repository.maven.internal.MavenPathParser;
+
+import com.google.common.base.Strings;
+
+/**
+ * Maven 2 path parser.
+ *
+ * @since 3.0
+ */
+public class Maven2MavenPathParser
+    implements MavenPathParser
+{
+  // TODO: TokenParser not capable to do this (yet)
+  // TODO: this is legacy, but here only to make it work
+  // TODO: consider moving/updating/fleshing out it from nx2 codebase, or improve TokenParser
```

Comment:
copy the code into internal package of this plugin for now, simplifies removal of legacy until probably a more proper solution to fix token-parser is done